Use color styles on the editor even if the classes were not set

jorgefilipecosta opened the pull request:
## Description
Fixes: https://github.com/WordPress/gutenberg/issues/6585
Before the editor just used the same classes on the frontend to apply color styles. If the classes were not set the editor would not apply them.

While for frontend view the current behaviour is the expected, on the editor is the themes passed us the color value we should use it even if the class was not created. 


## How has this been tested?
Add a palette that sets a color without implementing the class, verify the color was used in the editor.
e.g:
```
		add_theme_support( 'editor-color-palette',
			array(
				'name' => 'test red',
				'color' => '#f00',
			)
		);
```

jorgefilipecosta committed:
Use color styles on the editor even if the classes were not set

gziolo commented:
Code wise looks good. Travis in unhappy though. 

Can you share the link to the previous setup for reference, the one which uses a class name?

jorgefilipecosta commented:
> Code wise looks good. Travis in unhappy though.

I rebased and the tests are now passing, I'm not certain why travis was unhappy.

> Can you share the link to the previous setup for reference, the one which uses a class name?

I'm not certain what you mean by "previous setup" the only change we are doing in this PR is that before if we had a class for a color, on the editor we just used the class without setting an inline style for the color. Right now, even if a color class exists we set an inline style anyway so we are not dependent on the theme providing a class to correctly show the colors on the editor. 
Before we did:
` backgroundColor: backgroundColor.class ? undefined : backgroundColor.value `
(we added the inline style only if the class was not set)
Now we do: 
backgroundColor: backgroundColor.value,
(always add the inline style)

gziolo reviewed (approved):
Okey, let's get it in 👍
